tools/unicode_properties_parse/format_width_estimate_intervals.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# The following code generates data for _Width_estimate_intervals_v2 in <format>.
+
+import re
+from dataclasses import dataclass
+from typing import TextIO
+from pathlib import Path
+
+
+class width_u:
+    is_1: bool = False
+    is_2: bool = True
+
+
+class table_u:
+    # A valid Unicode code point won't exceed max_u.
+    max_u: int = 0x10FFFF
+
+    def __init__(self):
+        self.table = [width_u.is_1] * (self.max_u + 1)
+
+    def fill_range(self, rng: tuple, width: bool):
+        from_, to_ = rng[0], rng[1]
+        assert from_ <= to_, "impl assertion failed"
+        assert to_ <= self.max_u, "impl assertion failed"
+        self.table[from_ : to_ + 1] = [width] * (to_ - from_ + 1)
+
+    def print_intervals(self):
+        # Print table for _Width_estimate_intervals_v2.
+        c = 0

```

Comment:
it'd be nice if this were a more descriptive name; no change requested though since it was pre-existing (i feel this way generally about this file)